components/PeoplePageView/index.tsx

```diff
@@ -0,0 +1,167 @@
+import s from './style.module.css'
+import DepartmentTree from 'components/DepartmentTree'
+import {
+	findParentDepartmentsOfDepartment,
+	findSubDepartmentsOfDepartment,
+} from 'pages/people/people.utils'
+import PersonCard from 'components/PersonCard'
+import { FaCheck, FaMagnifyingGlass } from 'react-icons/fa6'
+import {
+	AVATAR_FILTER_LABEL,
+	DEPARTMENT_FILTER_TITLE,
+	NO_RESULTS,
+	PEOPLE_PAGE_SUBTITLE,
+	PEOPLE_PAGE_TITLE,
+	SEARCH_PEOPLE_PLACEHOLDER,
+} from '../constants'
+
+interface Props {
+	allPeople: PersonRecord[]
+	allDepartments: DepartmentRecord[]
+}
+
+export default function PeoplePageView({
+	allPeople,
+	allDepartments,
+}: Props): React.ReactElement {
+	// state variables
+	const [filteredPeople, setFilteredPeople] =
+		useState<PersonRecord[]>(allPeople)
+	const [selectedDepartment, setSelectedDepartment] =
+		useState<DepartmentRecord | null>(null)
+	const [childrenOfSelectedDepartment, setChildrenOfSelectedDepartment] =
+		useState<DepartmentRecord[] | null>([])
+	const [parentsOfSelectedDepartment, setParentsOfSelectedDepartment] =
+		useState<DepartmentRecord[] | null>([])
+	const [hidePeopleWithoutAvatar, setHidePeopleWithoutAvatar] =
+		useState<boolean>(false)
+	const [searchText, setSearchText] = useState<string>('')
+
+	// handlers
+	const handleSelectDepartment = useCallback(
+		(d: DepartmentRecord) => {
+			setSelectedDepartment(d)
+			setChildrenOfSelectedDepartment(findSubDepartmentsOfDepartment(d))
+			setParentsOfSelectedDepartment(
+				findParentDepartmentsOfDepartment(d, allDepartments)
+			)
+		},
+		[
+			setSelectedDepartment,
+			setChildrenOfSelectedDepartment,
+			setParentsOfSelectedDepartment,
+			allDepartments,
+		]
+	)
+
+	const handleToggleAvatars = useCallback(() => {
+		setHidePeopleWithoutAvatar(!hidePeopleWithoutAvatar)
+	}, [hidePeopleWithoutAvatar])
+
+	const handleTyping = useCallback((val: string) => {
+		setSearchText(val)
+	}, [])
+
+	// functionality
+	const applyFilters = useCallback(() => {
+		let newFiltered = [...allPeople]
+		if (hidePeopleWithoutAvatar) {
+			newFiltered = newFiltered.filter((p: PersonRecord) => p.avatar?.url)
+		}
+		// filter by selected department and any possible sub departments
+		if (selectedDepartment) {
+			const names = [selectedDepartment.name]
+			if (childrenOfSelectedDepartment.length > 0) {
+				names.push(
+					...childrenOfSelectedDepartment.map((d: DepartmentRecord) => d.name)
+				)
+			}
+			newFiltered = newFiltered.filter((p: PersonRecord) =>
+				names.includes(p.department.name)
+			)
+		}
+		if (searchText.length > 0) {
+			newFiltered = newFiltered.filter((p: PersonRecord) =>
+				p.name.toLowerCase().includes(searchText.toLowerCase())
+			)
+		}
+		setFilteredPeople(newFiltered)
+	}, [
+		allPeople,
+		hidePeopleWithoutAvatar,
+		selectedDepartment,
+		searchText,
+		childrenOfSelectedDepartment,
+	])
+
+	useEffect(() => {
+		applyFilters()
+		// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I know what you're saying. I thought perhaps it would look cleaner this way, but I do agree that there's risk with disabling that rule. Ultimately, this is where PR feedback would come in